sklearn/utils/_test_common/instance_generator.py

```diff
@@ -484,6 +484,20 @@
     Birch: {"check_dict_unchanged": dict(n_clusters=1)},
     BisectingKMeans: {"check_dict_unchanged": dict(max_iter=5, n_clusters=1, n_init=2)},
     CCA: {"check_dict_unchanged": dict(max_iter=5, n_components=1)},
+    DecisionTreeRegressor: {
+        "check_sample_weights_invariance": [
+            dict(criterion="squared_error"),
+            dict(criterion="absolute_error"),
+            dict(criterion="friedman_mse"),
+            dict(criterion="poisson"),
+        ],
+        "check_unit_sample_weights": [
+            dict(criterion="squared_error"),
+            dict(criterion="absolute_error"),
+            dict(criterion="friedman_mse"),
+            dict(criterion="poisson"),
+        ],
+    },
```

Comment:
We should probably do the same for the `criterion` param of `DecisionTreeClassifier` and the `loss` parameter of `HistGradientBoostingClassifier`/`Regressor`.

For `QuantileRegressor` we might want to test for a `quantile` parameter value that results in a non-symmetric loss (e.g. `quantile=0.75`).

We could also do the same for the `solver` parameter of `LogisticRegression` and maybe a few other important models with a `solver` parameter.

Reply:
Sure! I have added `Tweedie/Quantile/Poisson/GammaRegressor` and `LogisticRegression` and `DecisionTreeClassifier`. For `HistGradientBoostingClassifier/Regressor` which are subsampling I am waiting for the new statistical test.

Reply:
For `HistGradientBoostingClassifier/Regressor`, we need to check, but I have the feeling that for small datasets and specific choices of hyperparameters we should be able to get a deterministic fit behavior and therefore be able to use the deterministic repetitions/weights equivalence check.